src/include/miopen/reduce_common.hpp

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2020 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_REDUCE_COMMON_HPP
+#define GUARD_MIOPEN_REDUCE_COMMON_HPP
+
+#include <half.hpp>
+#include <miopen/bfloat16.hpp>
+
+namespace reduce {
+
+// data type conversion
+template <typename T>
```

Comment:
Please see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/246/files#r442793081
Any feedback?

Reply:
Yes,  good suggestion. 
Do you mean to use a function template with two parameters rather than using a functional class with operator() ? 

I am referring to the codes in 
src/kernels/composable_kernel/include/utility/float_type.hpp

I will make change fas miopen/reduce_common.hpp is for the host-side using.

Reply:
Yes, good suggestion.
Do you mean to use a function template with two parameters rather than using a functional struct with operator() ?

I am actually referring to the codes in
src/kernels/composable_kernel/include/utility/float_type.hpp

I will make change as miopen/reduce_common.hpp is for the host-side using.

Reply:
It's up to you. I just wanted to make sure you've seen my comment.